waku/v2/protocol/waku_message.nim

```diff
@@ -23,7 +24,7 @@ type
     # the proof field indicates that the message is not a spam
     # this field will be used in the rln-relay protocol
     # XXX Experimental, this is part of https://rfc.vac.dev/spec/17/ spec and not yet part of WakuMessage spec
-    proof*: seq[byte]
+    proof*: NonSpamProof
```

Comment:
NonSpamProof...maybe we can have a different name?

rlnProof
NotSpamProof
rateLimitProof